synapse/storage/databases/main/transactions.py

```diff
@@ -560,3 +560,13 @@ def get_destinations_paginate_txn(
         return await self.db_pool.runInteraction(
             "get_destinations_paginate_txn", get_destinations_paginate_txn
         )
+
+    async def is_destination(self, destination: str) -> Optional[bool]:
+        """Function to check if a destination is a destination of this server."""
```

Comment:
I think this could probably return just a `bool` since in this case `None` means `False`?

I'm also not sure if this name is clear, but I'm not thinking of a better one at the moment... maybe `does_destination_exist` or `is_destination_known`?

Also the docstring isn't really clear, maybe:

> Check if a destination is known to the server.